homeassistant/components/energy/data.py

```diff
@@ -0,0 +1,132 @@
+"""Energy data."""
+from __future__ import annotations
+
+from typing import Optional, TypedDict, cast
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import singleton, storage
+
+from .const import DOMAIN
+
+STORAGE_VERSION = 1
+STORAGE_KEY = DOMAIN
+
+
+HOME_CONSUMPTION_SCHEMA = vol.Schema(
+    {
+        vol.Required("stat_consumption"): str,
+        vol.Required("stat_cost"): vol.Any(None, str),
```

Comment:
```suggestion
        vol.Required("stat_cost"): vol.Any(str, None),
```

Reply:
Is there any reason why you would prefer this order?

Reply:
It's more clear to start with the type of the value and add the missing value type at the end, similar to how we do typing. We also want to be consistent across the module.

Reply:
The way `vol.Any` works is that it tries all options and the first one that doesn't raise is picked. Any callable (`str`, `vol.Coerce(float)`) will be called to parse it. If we specify `None` first, it's not a callable and it will just be `is` checked. So it's slightly faster to put `None` first, as you avoid calling a function – but it's also probably neglectible. I've pushed an update to change it.

Reply:
Actually, that last statement is not correct and now I have no idea how vol.Any works 

```
>>> print(vol.Schema(vol.Any(bool, None))(None))
None
>>> print(vol.Schema(vol.Any(None, bool))(None))
None
```